plugin/completion.py

```diff
@@ -125,3 +99,29 @@ def translated_regions(self, edit_region: sublime.Region) -> Generator[sublime.R
             translation = region.b - primary_cursor_position
             translated_edit_region = sublime.Region(edit_region.a + translation, edit_region.b + translation)
             yield translated_edit_region
+
+    def epilogue(self, item: CompletionItem, session_name: str) -> None:
+        session = self.session_by_name(session_name, 'completionProvider.resolveProvider')
+        if session:
+            request = Request.resolveCompletionItem(item, self.view)
+            session.send_request_async(request, lambda response: self.on_resolved(response, session_name))
+        else:
+            self.on_resolved(item, session_name)
+
+    def on_resolved(self, item: CompletionItem, session_name: str) -> None:
+        def run_main() -> None:
```

Comment:
This `run_main` doesn't really belong here. The expectation is that this function is called on the main thread but it shouldn't be the responsibility of that function to do that.

I would expect something likes this rather, in the epilogue:

```py
            def resolve_on_main_thread(item: CompletionItem, session_name: str) -> None:
                sublime.set_timeout(lambda: self.on_resolved(item, session_name))

            request = Request.resolveCompletionItem(item, self.view)
            session.send_request_async(request, lambda response: resolve_on_main_thread(response, session_name))
```